hubbard/hamiltonian.py

```diff
@@ -514,8 +560,10 @@ def iterate(self, calc_n_method, q=None, mixer=None, **kwargs):
         self.update_hamiltonian()
 
         # Store total energy
-        self.Etot = Etot - self.U * (ni[0]*ni[-1]).sum()
-
+        self.Etot = Etot - (self.U * ni[0]*ni[-1]).sum()
+        # Inter-orbital term
+        if self.Uij is not None:
+            self.Etot -= 0.5*self.Uij @ (ni[0]+ni[-1]) @ (ni[0]+ni[-1])
```

Comment:
Where does the "extra" factor `(ni[0]+ni[-1])` come from?

Reply:
Comes from the constant term obtained from the MF approximation for the generalized Hubbard Hamiltonian, that's why is added to the energy. As you can see, for the diagonal case (U) the constant term is also quadratic on the spin density. The same happens here, but the repulsion between atoms is between both spin components, which makes sense because there is no Pauli exclusion between atoms.

Reply:
All clear now, thanks! (I somehow missed the quadratic form for the diagonal term).